pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -46,19 +47,31 @@ func (rs *RevisionSpec) Validate() *apis.FieldError {
 	if equality.Semantic.DeepEqual(rs, &RevisionSpec{}) {
 		return apis.ErrMissingField(apis.CurrentField)
 	}
-	errs := validateContainer(rs.Container).ViaField("container").
-		Also(validateBuildRef(rs.BuildRef).ViaField("buildRef"))
 
-	if err := rs.DeprecatedConcurrencyModel.Validate().ViaField("concurrencyModel"); err != nil {
-		errs = errs.Also(err)
-	} else if err := ValidateContainerConcurrency(rs.ContainerConcurrency, rs.DeprecatedConcurrencyModel); err != nil {
-		errs = errs.Also(err)
+	volumes := sets.NewString()
+	var errs *apis.FieldError
+	for i, volume := range rs.Volumes {
+		if volumes.Has(volume.Name) {
+			errs = errs.Also((&apis.FieldError{
```

Comment:
You should use my new method here ErrInvalidArrayValue :-)

Reply:
Hmm, as written I could use it, but that's a bug.  I should have `"name"` instead of `apis.CurrentField` below, which makes it so I cannot :(

Reply:
You also made me realize I didn't add testing for the `VolumeMount` logic in the `RevisionSpec` `Validate()` method.  thanks!